metrics/api/src/main/java/io/helidon/metrics/api/MetricStore.java

```diff
@@ -556,6 +556,14 @@ private <S> S access(Lock lock, Callable<S> action) {
         }
     }
 
+    private static boolean tagsMatch(Tag[] tags, Map<String, String> tagMap) {
+        Map<String, String> newTags = new TreeMap<>();
```

Comment:
I was wondering if this should optimize for the case of a single tag (to avoid creating the Map), but now I see you mention in the description that this is an infrequent operation. So LGTM!